app/src/main/java/org/mozilla/focus/browser/VideoVoiceCommandMediaSession.kt

```diff
@@ -233,32 +236,25 @@ class VideoVoiceCommandMediaSession @UiThread constructor(
             |}
             """.trimMargin()
 
-        override fun onPlay() {
-            onPlayPause() // See class javadoc for details.
-        }
-
-        override fun onPause() {
-            onPlayPause() // See class javadoc for details.
-        }
-
-        private fun onPlayPause() {
-            // Due to time constraints, this code is written for a single video on the page,
-            // which should cover the majority use case.
-            //
-            // We don't handle audio: see class javadoc for details.
+        // NB: this hasn't been tested on expressions that are more than one line.
+        private fun evalJSWithTargetVideo(getExpressionToEval: (videoId: String) -> String) {
+            val expressionToEval = getExpressionToEval(ID_TARGET_VIDEO)
             webView?.evalJS("""
                 |(function() {
                 |    $GET_TARGET_VIDEO_OR_RETURN
-                |
-                |    if ($ID_TARGET_VIDEO.paused) {
-                |        $ID_TARGET_VIDEO.play();
-                |    } else {
-                |       $ID_TARGET_VIDEO.pause();
-                |   }
+                |    $expressionToEval
                 |})();
                 """.trimMargin())
         }
 
+        override fun onPlay() {
+            evalJSWithTargetVideo { videoId -> "$videoId.play();" }
```

Comment:
You don't think you need to check for paused() state anymore? What happens when people say the same thing twice "pause....pause" does it still work?

Reply:
No, I don't think we need to check: the API is designed to handle this.

Voice commands will do what you say. Calling a redundant `play/pause` on a DOM video will be a no-op.

If it's play/pause media button, it'll send the appropriate play/pause based on the current Java playback state, which we've already synced from JS.

I'll add a comment to explain this.